tutorials/src/cpu/detector/detector_ray_scan.cpp

```diff
@@ -21,28 +22,40 @@
 #include <vecmem/memory/host_memory_resource.hpp>
 
 // System include(s)
+#include <fstream>
 #include <iostream>
+#include <string>
 
 // Hash of the "correct" geometry
 constexpr std::size_t root_hash = 3244;
 
-/// Check a given detecor for consistent linking by shooting rays/helices and
-/// recording every intersection with the geometry. This intersection record
-/// can then be checked for matching portals at the volume boundary surfaces
-/// ( @c trace_intersections ) and checked for a consistent 'path' from volume
-/// to volume ( @c check_consistency ). See also documentation in
-/// 'tests/common/tools/ray_scan_utils.hpp'.
-int main() {
+/// Execution function for the ray scan
+///
+/// @tparam detector_t the type of the detector (toy, read in)
+/// @param det the detector object
+/// @param theta_steps number of steps in theta
+/// @param phi_steps number of steps in phi
+///
+/// @return 1 if the scan was successful, 0 otherwise
+template <typename detector_t>
+int run_ray_scan(const detector_t &det, unsigned int theta_steps = 100u,
+                 unsigned int phi_steps = 100u,
+                 const std::string &outfile = "") {
+
+    std::fstream out;
```

Comment:
This should probably use the detray ```file_handle```, instead of openening an ```fstream```, since it does some automatic checking. You could try something like this:
```suggestion
    detray::io::detail::file_handle file{outfile, ".csv", std::ios::out};
    *file << "ray_index,x,y,z," << std::endl; 
```
And then use ```*file``` in place of ```out```

Reply:
Should we also add the trunc flag (```std::ios::out | std::ios::trunc```) to replace the file?